src/renderer/api/catalog-entity-registry.ts

```diff
@@ -0,0 +1,61 @@
+import { action, observable } from "mobx";
+import { broadcastMessage, subscribeToBroadcast } from "../../common/ipc";
+import { CatalogCategory, CatalogEntity, CatalogEntityData } from "../../common/catalog-entity";
+import { catalogCategoryRegistry, CatalogCategoryRegistry } from "../../common/catalog-category-registry";
+import "../../common/catalog-entities";
+
+export class CatalogEntityRegistry {
+  @observable protected _items: CatalogEntity[] = observable.array([], { deep: true });
+
+  constructor(private categoryRegistry: CatalogCategoryRegistry) {}
+
+  init() {
+    subscribeToBroadcast("catalog:items", (ev, items: CatalogEntityData[]) => {
+      this.updateItems(items);
+    });
+    broadcastMessage("catalog:broadcast");
+  }
+
+  @action updateItems(items: CatalogEntityData[]) {
```

Comment:
This seems contrary to the ethos of having multiple sources. Shouldn't the registry just combind the arrays that it is given, and let the sources deal with editing?

Reply:
Plus we really shouldn't care about ordering here. That should be a display problem and we should have a uniform sorting for display.